### PR TITLE
SED-2968-Support charts with 2 axis

### PR DESCRIPTION
**Pull request by lucianbaciu**

### CODE REVIEW

File: projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet-settings/chart-dashlet-settings.component.html [hunk 1]
```diff
@@ -35,6 +34,19 @@
...
         </div>
+        <div class="form-row">
+          <step-form-field>
+            <step-label>Secondary Axes Aggregation</step-label>
+            <mat-select
+              [value]="item.chartSettings!.secondaryAxes?.aggregation"
+              (valueChange)="onSecondaryAggregateSelect($event)"
+            >
+              <mat-option [value]="undefined">-</mat-option>
+              <mat-option *ngFor="let attr of AGGREGATE_TYPES" [value]="attr">{{ attr }} </mat-option>
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet-settings/chart-dashlet-settings.component.html:45]:
```suggestion
    @for(attr of AGGREGATE_TYPES; track attr) {
              <mat-option [value]="attr">{{ attr }} </mat-option>
    }
```

File: projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet-settings/chart-dashlet-settings.component.html [hunk 2]
```diff
@@ -35,6 +34,19 @@
...
+            </mat-select>
+          </step-form-field>
+          <div></div>
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet-settings/chart-dashlet-settings.component.html:48]: What is the purpose of this empty `<div></div>`?

File: projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet-settings/chart-dashlet-settings.component.ts [hunk 3]
```diff
@@ -66,4 +94,12 @@ export class ChartDashletSettingsComponent implements OnInit {
...
     this._dialogRef.close({ ...this.item });
   }
+
+  switchAggregate(aggregate: ChartAggregation) {
+    this.item.chartSettings!.primaryAxes.aggregation = aggregate;
+  }
+
+  get ChartAggregation() {
+    return ChartAggregation;
+  }
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet-settings/chart-dashlet-settings.component.ts:104]:
```suggestion
readonly ChartAggregation = ChartAggregation;
```

File: projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet/chart-dashlet.component.ts [hunk 4]
```diff
@@ -60,9 +71,9 @@ export class ChartDashletComponent implements OnInit {
   @Output() shiftRight = new EventEmitter();
 
   readonly PCL_VALUES = [80, 90, 99];
-  selectedPclValue: number = this.PCL_VALUES[0];
+  selectedPclValue: number | undefined;
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet/chart-dashlet.component.ts:74]:
```suggestion
  selectedPclValue?: number;
```

File: projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet/chart-dashlet.component.ts [hunk 5]
```diff
@@ -397,4 +458,8 @@ export class ChartDashletComponent implements OnInit {
   private getSeriesKeys(attributes: BucketAttributes, groupDimensions: string[]): (string | undefined)[] {
     return groupDimensions.map((field) => attributes?.[field]);
   }
+
+  get ChartAggregation() {
+    return ChartAggregation;
+  }
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet/chart-dashlet.component.ts:464]:
```suggestion
  readonly ChartAggregation = ChartAggregation;
```

**Review by dvladir — CHANGES_REQUESTED** [review]

File: projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet-settings/chart-dashlet-settings.component.html (hunk 2 above)
Comment (lucianbaciu) [reply projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet-settings/chart-dashlet-settings.component.html:50]: to force the parent container to split into 2 equal elements

**Review by dvladir — APPROVED** [review]

**Review by dvladir — CHANGES_REQUESTED** [review]
@lucianbaciu Please check last remarks in the ticket

File: projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet/chart-dashlet.component.ts [hunk 6]
```diff
@@ -83,17 +93,29 @@ export class ChartDashletComponent implements OnInit {
...
-    if (this.selectedAggregate === ChartAggregation.PERCENTILE) {
+    this.prepareState(this.item);
+    this.fetchDataAndCreateChart().subscribe();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['item']) {
+      this.prepareState(this.item);
+      this.refresh(true).subscribe();
+    }
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet/chart-dashlet.component.ts:104]:
```suggestion
    const cItem = changes['item'];
    if (cItem?.previousValue !== cItem?.currentValue || cItem.firstChange)
      this.prepareState(cItem.currentValue);
      this.refresh(true).subscribe();
    }
```

**Review by dvladir — CHANGES_REQUESTED** [review]

**Review by dvladir — APPROVED** [review]

File: projects/step-frontend/src/lib/modules/timeseries/components/dashboard/dashboard.component.ts [hunk 7]
```diff
@@ -194,6 +194,7 @@ export class DashboardComponent implements OnInit, OnDestroy {
 
   saveEditChanges() {
     this.editMode = false;
+    console.log(this.dashboard);
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/components/dashboard/dashboard.component.ts:197]:
```suggestion
```

**Review by dvladir — APPROVED** [review]

**Review by dvladir — APPROVED** [review]

**Pull request merged**